File: card-core/src/main/java/com/jean202/cardmizer/core/domain/Card.java
package com.jean202.cardmizer.core.domain;

import java.util.Objects;

public record Card(
        CardId id,
        String issuerName,
        String productName,
        CardType cardType,
        int priority
) {
    public Card {
        Objects.requireNonNull(id, "id must not be null");
        cardType = Objects.requireNonNull(cardType, "cardType must not be null");

        if (issuerName == null || issuerName.isBlank()) {
            throw new IllegalArgumentException("Issuer name must not be blank");
        }
        if (productName == null || productName.isBlank()) {
            throw new IllegalArgumentException("Product name must not be blank");
        }
        if (priority < 1) {
            throw new IllegalArgumentException("Priority must start from 1");
        }
    }

    public String displayName() {
        return issuerName + " " + productName;
    }
}

File: card-core/src/main/kotlin/com/jean202/cardmizer/core/domain/Card.kt
package com.jean202.cardmizer.core.domain

data class Card(
    val id: CardId,
    val issuerName: String,
    val productName: String,
    val cardType: CardType,
    val priority: Int,
) {
    init {
        require(issuerName.isNotBlank()) { "Issuer name must not be blank" }
        require(productName.isNotBlank()) { "Product name must not be blank" }
        require(priority >= 1) { "Priority must start from 1" }
    }

    fun displayName(): String = "$issuerName $productName"
}
